### PR TITLE
Enhance slow-propose test configuration

### PR DESCRIPTION
**Pull request by sergefdrv**

### CODE REVIEW

File: pkg/deploytest/deployment.go [hunk 1]
```diff
@@ -128,13 +128,19 @@ func NewDeployment(testConfig *TestConfig) (*Deployment, error) {
...
-		if i == 0 {
-			issConfig = testConfig.FirstReplicaISSConfig
+		// ISS configuration
+		issConfig := iss.DefaultConfig(membership)
+		if testConfig.SlowProposeReplicas[i] {
+			// Increase MaxProposeDelay such that it is likely to trigger view change by segment timeout
+			issConfig.MaxProposeDelay = (issConfig.PBFTViewChangeSegmentTimeout + issConfig.MaxProposeDelay) / time.Duration(issConfig.SegmentLength)
+
+			// Make sure it also has change to trigger view change by batch timeout
+			if issConfig.MaxProposeDelay < issConfig.PBFTViewChangeBatchTimeout {
```
Comment (matejpavlovic) [inline pkg/deploytest/deployment.go:138]: In fact, doesn't this obviate the previous lines? At the end of the day, it is always the batch timeout that is triggered, making the segment timeout irrelevant. It looks like simply always setting `MaxProposeDelay` to the batch timeout has the same effect in practice.
One could argue that if the `MaxProposeDelay` is *exactly* the same as the batch timeout, the view change might or might not be triggered, but in practice, it almost always does, making the segment timeout extremely unlikely, especially with segments that are not trivially short.
Bottom line: what about just
```go
if testConfig.SlowProposeReplicas[i] {
    issConfig.MaxProposeDelay = issConfig.PBFTViewChangeBatchTimeout
}
```

File: pkg/deploytest/deployment.go [hunk 2]
```diff
@@ -128,13 +128,19 @@ func NewDeployment(testConfig *TestConfig) (*Deployment, error) {
...
-		var issConfig *iss.Config
-		if i == 0 {
-			issConfig = testConfig.FirstReplicaISSConfig
+		// ISS configuration
+		issConfig := iss.DefaultConfig(membership)
+		if testConfig.SlowProposeReplicas[i] {
+			// Increase MaxProposeDelay such that it is likely to trigger view change by segment timeout
+			issConfig.MaxProposeDelay = (issConfig.PBFTViewChangeSegmentTimeout + issConfig.MaxProposeDelay) / time.Duration(issConfig.SegmentLength)
+
+			// Make sure it also has change to trigger view change by batch timeout
```
Comment (matejpavlovic) [inline pkg/deploytest/deployment.go:137]: Typo

File: pkg/deploytest/deployment.go (hunk 1 above)
Comment (sergefdrv) [reply pkg/deploytest/deployment.go:138]: The logic was to make it trigger the segment timeout, but ensure that batch timeout can also be triggered, without making any assumption about the default timeout values.

Comment (matejpavlovic) [reply pkg/deploytest/deployment.go:138]: But the way it is now, it almost never triggers the segment timeout in practice, only the batch timeout.

Comment (sergefdrv) [reply pkg/deploytest/deployment.go:138]: It happens that the following holds in the current default configuration: 
```
PBFTViewChangeSegmentTimeout < SegmentLength * PBFTViewChangeBatchTimeout
```
but I didn't want to assume this.

Comment (sergefdrv) [reply pkg/deploytest/deployment.go:138]: > One could argue that if the `MaxProposeDelay` is exactly the same as the batch timeout, the view change might or might not be triggered, but in practice, it almost always does, making the segment timeout extremely unlikely, especially with segments that are not trivially short.

This may not be necessarily true when we switch into using simulated network and local clocks.

Comment (matejpavlovic) [reply pkg/deploytest/deployment.go:138]: If that didn't hold, the segment timeout would become almost irrelevant. In such a case, if a node didn't commit everything by the segment timeout, it inevitably must have observed at least one batch timeout and already entered view change (modulo, perhaps, some corner cases around segment-level checkpoints).

Comment (sergefdrv) [reply pkg/deploytest/deployment.go:138]: > if a node didn't commit everything by the segment timeout, it inevitably must have observed at least one batch timeout

Oh, that observation actually makes a good case for setting `MaxProposeDelay = PBFTViewChangeBatchTimeout` :+1:

**Comment by sergefdrv** [comment]
Summary of [changes](https://github.com/filecoin-project/mir/compare/5921ee6d13b2e111261e43eef492c545e13f5de7..9642503b429db150505db6cc536e98761a390156): simplified slow config to setting `MaxProposeDelay = PBFTViewChangeBatchTimeout`

**Review by matejpavlovic — APPROVED** [review]
👍
